hudi-client/hudi-client-common/src/main/java/org/apache/hudi/async/HoodieAsyncService.java

```diff
@@ -125,6 +125,10 @@ public void shutdown(boolean force) {
    * @param onShutdownCallback
    */
   public void start(Function<Boolean, Boolean> onShutdownCallback) {
+    if (started) {
+      LOG.warn("The async service already started.");
+      return;
+    }
```

Comment:
Wondering when the caller invoked the service instant twice !

Reply:
I don't see misusing scenarios from the codebase but it's a potential risk, esp. for users run separate service jobs.

Reply:
Thanks for the explanation ~

Reply:
Hmm, this has caused a bug, we should reset the flag in when the future complete.